components/bt/bluedroid/btc/profile/esp/blufi/blufi_prf.c

```diff
@@ -370,7 +370,9 @@ static void btc_blufi_recv_handler(uint8_t *data, int len)
             && (blufi_env.cbs && blufi_env.cbs->checksum_func)) {
         checksum = blufi_env.cbs->checksum_func(hdr->seq, &hdr->seq, hdr->data_len + 2);
         if (memcmp(&checksum, &hdr->data[hdr->data_len], 2) != 0) {
-            LOG_ERROR("%s checksum error %04x, pkt %04x\n", __func__, checksum, *(uint16_t *)&hdr->data[hdr->data_len]);
+            uint16_t checksum_pkt = hdr->data[hdr->data_len] |
+                                    (((uint16_t) hdr->data[hdr->data_len + 1]) << 8);
+            LOG_ERROR("%s checksum error %04x, pkt %04x\n", __func__, checksum, checksum_pkt);
```

Comment:
I think if LOG_ERROR is disabled this is going to give an unused variable warning. :(

How about calculating checksum_pkt outside the if statement, and replacing the memcmp with a variable compare?